pinot-core/src/main/java/org/apache/pinot/server/realtime/ServerSegmentCompletionProtocolHandler.java

```diff
@@ -64,6 +65,7 @@ public static void init(PinotConfiguration uploaderConfig) {
       _sslContext = new ClientSSLContextGenerator(httpsConfig.subset(CommonConstants.PREFIX_OF_SSL_SUBSET)).generate();
       _controllerHttpsPort = httpsConfig.getProperty(CONFIG_OF_CONTROLLER_HTTPS_PORT, Integer.class);
     }
+    _protocol = uploaderConfig.getProperty(CONFIG_OF_PROTOCOL, HTTP_PROTOCOL);
```

Comment:
Do we need this extra config? If https is enabled, can we directly set https as the scheme?

Reply:
unfortunately, the way `ServerSegmentCompletionProtocolHandler` is written and used, there's no sane way to pass "non-specific" configs. Afaik LinkedIn has some internal dependencies on this behavior, so I can't change it either.

Reply:
I don't quite follow the logic here. Is it possible that we don't enable `CONFIG_OF_CONTROLLER_HTTPS_ENABLED` but config the protocol to be `HTTPS`?
In line 198, when `_controllerHttpsPort` is set, it automatically use `HTTPS`

Reply:
`CONFIG_OF_CONTROLLER_HTTPS_ENABLED` really is a feature flag that configures a hand-rolled TLS key and truststore. it also requires a dedicated https port (there may be only one)